pending-documentation.md

```diff
@@ -125,20 +195,37 @@ This generates two heating rules (where the hole syntaxes `"[]" "+" AExp` and
 `AExp "+" "[]"` is automatically added to create an evaluation context):
 
 ```k
-rule <k> AE1:AExp + AE2:AExp => AE1 ~> [] + AE2 ... </k> requires notBool isKResult(AE1)
-rule <k> AE1:AExp + AE2:AExp => AE2 ~> AE1 + [] ... </k> requires notBool isKResult(AE2)
+rule <k> HOLE:AExp +  AE2:AExp => HOLE ~>  [] + AE2 ... </k> [heat]
+rule <k>  AE1:AExp + HOLE:AExp => HOLE ~> AE1 +  [] ... </k> [heat]
 ```
 
 And two corresponding cooling rules:
 
 ```k
-rule <k> AE1:KResult ~> [] + AE2 => AE1 + AE2 ... </k>
-rule <k> AE2:KResult ~> AE1 + [] => AE1 + AE2 ... </k>
-```
-
-Note that these rules depend on the definition of `KResult`, so you need to
-define this for K. For instance, we tell K that a term is fully evaluated once it
-becomes an `Int` here:
+rule <k> HOLE:AExp ~>  [] + AE2 => HOLE +  AE2 ... </k> [cool]
+rule <k> HOLE:AExp ~> AE1 +  [] =>  AE1 + HOLE ... </k> [cool]
+```
+
+You will note that these rules can apply one after another infinitely. In
+practice, the `KResult` sort is used to break this cycle by ensuring that only
+terms that are not part of the `KResult` sort will be heated. In the theory of
```

Comment:
> In the theory of K, these rules represent equalities between structurally distinct states rather
than transitions in the transition system.

This is not presently true of the generated Kore.

Reply:
Did not know that, thanks. I will update.